### PR TITLE
Closes #44 Reservation confirmation / overview

### LINKED ISSUE #44: Extend reservation overview screen with channel information

### PR DESCRIPTION
**Pull request by nikola-mladenovic**
Currently, you can reach reservation confirmation screen from the my account page.

What needs to be added - parsing push notification payload, after which, in case of a reservation confirmation, this screen would also be presented.

### CODE REVIEW

File: Sunrise/ViewModels/OrdersViewModel.swift [hunk 1]
```diff
@@ -74,10 +74,20 @@ class OrdersViewModel: BaseViewModel {
...
     }
 
-    func orderOverviewViewModelForOrderAtIndexPath(indexPath: NSIndexPath) -> OrderOverviewViewModel {
-        let orderOverviewViewModel = OrderOverviewViewModel()
-        orderOverviewViewModel.order.value = indexPath.section == 0 ? orders[indexPath.row] : reservations[indexPath.row]
-        return orderOverviewViewModel
+    func orderOverviewViewModelForOrderAtIndexPath(indexPath: NSIndexPath) -> OrderOverviewViewModel? {
+        if indexPath.section == 0 {
+            let orderOverviewViewModel = OrderOverviewViewModel()
+            orderOverviewViewModel.order.value = indexPath.section == 0 ? orders[indexPath.row] : reservations[indexPath.row]
```
Comment (cneijenhuis) [inline Sunrise/ViewModels/OrdersViewModel.swift]: The check for `indexPath.section == 0` will always return true here. Should probably look like the function below.

**Comment by cneijenhuis** [comment]
Otherwise the code LGTM 👍

File: Sunrise/ViewModels/OrdersViewModel.swift (hunk 1 above)
Comment (nikola-mladenovic) [inline Sunrise/ViewModels/OrdersViewModel.swift]: Yes, good catch. That was a left over from before reservation view model was introduced 👍

**Comment by cneijenhuis** [comment]
For the view, I noticed a couple of minor differences compared to the mock:
- The Product Name is centered (should be left-aligned like the store name)
- The non-headline font color is black (should be some grey)
- The non-headline font feels bigger compared to the mock - I'm not sure if the Opening Hours will fit onto a smaller screen.
- The Map-View should have the same gap (to the left) as the text

Also, there is a larger gap between the Opening hours and the end of the screen that I'd shrink.

So some minor things left to do 😉

**Comment by nikola-mladenovic** [comment]
Incorporated:
- _The Product Name is centered (should be left-aligned like the store name)_
- _The non-headline font feels bigger compared to the mock._
  Note: I've reduced the font size used here on all devices, but in cause of a smaller device, i've set auto shrink option, so the size will be reduced so that the information first.
- _The Map-View should have the same gap (to the left) as the text_

Questions:
- _The non-headline font color is black (should be some grey)_
  Color used for all non-headline fonts is #353E48 which I picked from the mock provided by Diana.
- _Also, there is a larger gap between the Opening hours and the end of the screen that I'd shrink._
  Do we want to put in some hardcoded text description about the pickup here? Something similar to what's in the mock, so that the screen looks more lively. WDYT @cneijenhuis ?

So some minor things left to do 😉

**Comment by cneijenhuis** [comment]
Let's not have a text/description there, as no one can come up with something quickly 😉

**Pull request merged**